### DISCUSSION TITLE
[BUG] Password less servers with V+ impossible

### DISCUSSION THREAD

**Discussion by spannerman79** [mbround18/valheim-docker#293] [Help!] [answered]
To run a passwordless server a couple of things need to be done;

1. Make sure that you are using ValheimPlus (or modded in some way to bypass that "feaure")
2. `disableServerPassword=true` in `BepInEx/config/valheim_plus.cfg` (Using V+ for example)
3. The server **cannot** be public ( `-public 0`)

I have tested this out by creating a stock standard Valheim server (steamcmd etc), grabbing ValheimPlus directly from their github, extracting it, editing V+ config (only changed `disableServerPassword=true`), and ran it ( `./start_server_bepinex.sh -name "MyValheimPlusServer" -port 2456 -world "myworld" -public 0`

I was able to connect in without a password (Direct IP, using localhost/127.0.0.1) and didn't need to enter a password.

The issue is I cannot do this with the way Odin or whatever script checks are in place.  Even with the correct requirements done - Odin prevents the start of the server, or in facts push's a password start flag which no matter what the setting in V+ you will still need that _oh so hard to guess_ password :stuck_out_tongue: 

Suggestions:

I was thinking since the more recent mod update was pushed the **only** way to ensure that no password was defined and allowed Odin to contine would be to ensure that it was anything but `TYPE=Vanilla` and a **must**, `PUBLIC=0`. In fact even having `PUBLIC=1` even with the config value set in V+ the server will fail as if you do not have a password.

Edit:  Using latest docker image at creation of this issue;

```
docker pull mbround18/valheim
Using default tag: latest
latest: Pulling from mbround18/valheim
Digest: sha256:06eb40108fb9aef3bb3465c1b667e6897988ed04fca8aa1c76e396fa4e8a9313
Status: Image is up to date for mbround18/valheim:latest
docker.io/mbround18/valheim:latest
```

**Comment by mbround18** [comment, marked as answer]
This is because, currently, odin is overwriting configs. This is being worked on currently in an effort to it being resolved. Please see https://github.com/mbround18/valheim-docker/issues/272 for further information

**Reply by spannerman79** [reply]
Yes I _have_ seen and read - yet nothing in there about Odin force setting launch options.

And how is an actual bug report a forum subject? re: moved to Discussions.

**Reply by mbround18** [reply]
That would be my bot, i can toss it back to an issue and add a label

**Reply by mbround18** [reply]
Good report @spannerman79 , I have been chatting with @LovecraftianHorror about this on Discord and brainstorming ways we can solve it.

**Reply by mbround18** [reply]
has this been resolved?

**Reply by spannerman79** [reply]
Going through old discussions - yes.

Resolved.
